byway/src/main/java/com/google/sps/servlets/CreateTripServlet.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.sps.data.Trip;
+import com.google.sps.data.UserInfo;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/createtrip")
+public class CreateTripServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+
+  @Override
+  public void init() {
+    System.setProperty(
+        DatastoreServiceConfig.DATASTORE_EMPTY_LIST_SUPPORT, Boolean.TRUE.toString());
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Key tripKey = createTripEntity();
+    String keyString = KeyFactory.keyToString(tripKey);
+    addTripForUser(keyString);
+    String json = gson.toJson(KeyFactory.keyToString(tripKey));
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  /*
+   * Adds new Trip entity with empty properties
+   **/
+  public Key createTripEntity() {
```

Comment:
As with the `UserInfo`, you might want to make this a static method in the `Trip` class. That way all manipulation of the string property names of the trip entity's properties is encapsulated within the `Trip`. That will make it easier make changes in the future, since then you don't have to worry about changing string property names across the whole codebase, or the possibility of missing a typo of one of the property names in some other file.

Reply:
done